src/get-html.js

```diff
@@ -0,0 +1,100 @@
+/**
+ * Copyright 2015 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/** @type {Array<string>} */
+const excludedTags = ['amp-analytics', 'amp-experiment', 'amp-bind-state'];
+
+/**
+ * Returns content of HTML node
+ * @param {!string} selector - CSS selector of the node to take content from
+ * @param {!Array<string>} attrs - tag attributes to be left in the stringified HTML
+ * @return {string}
+ */
+export function getHtml(selector, attrs) {
+  const root = self.document.querySelector(selector);
+  const result = [];
+
+  if (root) {
+    appendToResult(root, attrs, result);
+  }
+
+  return result.join('').replace(/\s{2,}/g, ' ');
+}
+
+/**
+ * @param {!Element} node - node to take content from
+ * @param {!Array<string>} attrs - tag attributes to be left in the stringified HTML
+ * @param {!Array<string>} result
+ */
+function appendToResult(node, attrs, result) {
+  const stack = [];
+
+  if (!node) {
+    return result;
+  }
+
+  stack.push(node);
+
+  while (stack.length > 0) {
+    node = stack.pop();
+
+    if (typeof node === 'string') {
+      result.push(node);
+    } else if (node && node.nodeType === Node.TEXT_NODE) {
+      result.push(node.textContent);
+    } else if (isApplicableNode(node)) {
+      appendOpenTag(node, attrs, result);
+      stack.push(`</${node.tagName.toLowerCase()}>`);
```

Comment:
Why isn't this just being pushed into `result`? It's on the stack before the child nodes, so it'll just get there anyways.